Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Undo/PrefabUndoAddEntityAsOverride.cpp

```diff
@@ -0,0 +1,101 @@
+                PrefabInstanceUtils::GetRelativePathFromClimbedInstances(climbUpResult.m_climbedInstances, true);
+
+            const AZStd::string parentEntityAliasPathForPatch =
+                entityAliasPathPrefixForPatch + parentEntityAliasPath;
+
+            const AZStd::string parentEntityAliasPathInFocusedTemplate =
+                PrefabDomUtils::PathStartingWithInstances +
+                climbUpResult.m_climbedInstances.back()->GetInstanceAlias() +
+                parentEntityAliasPathForPatch;
+
+            PrefabDom parentEntityDomAfterAddingEntity;
+            m_instanceToTemplateInterface->GenerateDomForEntity(parentEntityDomAfterAddingEntity, parentEntity);
+
+            TemplateId focusedTemplateId = focusedInstance.GetTemplateId();
+            PrefabDom& focusedTemplateDom =
+                m_prefabSystemComponentInterface->FindTemplateDom(focusedTemplateId);
+            PrefabDomPath entityPathInFocusedTemplate(parentEntityAliasPathInFocusedTemplate.c_str());
+            // This scope is added to limit their usage and ensure DOM is not modified when it is being used.
+            {
+                // DOM value pointers can't be relied upon if the original DOM gets modified after pointer creation.
+                const PrefabDomValue* parentEntityDomInFocusedTemplate = entityPathInFocusedTemplate.Get(focusedTemplateDom);
+                AZ_Assert(parentEntityDomInFocusedTemplate,
+                    "Could not load parent entity's DOM from the focused template's DOM. "
+                    "Focused template id: '%llu'.", static_cast<AZ::u64>(focusedTemplateId));
+
+                PrefabUndoUtils::GenerateUpdateEntityPatch(m_redoPatch,
+                    *parentEntityDomInFocusedTemplate, parentEntityDomAfterAddingEntity, parentEntityAliasPathForPatch);
+            }
+
+            const AZStd::string newEntityAliasPathForPatch =
+                entityAliasPathPrefixForPatch + newEntityAliasPath;
+
+            PrefabDom newEntityDom;
+            m_instanceToTemplateInterface->GenerateDomForEntity(newEntityDom, newEntity);
+            PrefabUndoUtils::GenerateAddEntityPatch(m_redoPatch, newEntityDom, newEntityAliasPathForPatch);
+
+            const LinkId linkId = climbUpResult.m_climbedInstances.back()->GetLinkId();
+            PrefabUndoUpdateLink::Capture(m_redoPatch, linkId);
+            
+            // Preemptively updates the cached DOM to prevent reloading instance DOM.
+            PrefabDomReference cachedOwningInstanceDom = owningInstance.GetCachedInstanceDom();
+            if (cachedOwningInstanceDom.has_value())
+            {
+                PrefabUndoUtils::UpdateCachedOwningInstanceDom(cachedOwningInstanceDom,
+                    parentEntityDomAfterAddingEntity, parentEntityAliasPath);
+                PrefabUndoUtils::UpdateCachedOwningInstanceDom(cachedOwningInstanceDom,
+                    newEntityDom, newEntityAliasPath);
+            }
```

Comment:
I think "owning" shouldn't be in the names here. Currently it is updating the owning instance but there shouldn't be any changes due to overrides?

For overrides, we should update the focused instance's cached instance DOM and the paths should be "/Instances/skipTopInstance" + override patch path. So the util function can be generic to handle both cases.

Reply:
oh never mind. I think updating the cached owning instance DOM is right here. The update is specific to this instance object. I'll resolve the comment.

Reply:
I tried that at beginning but it will cause newly added entities go under unrelated instance during undo/redo.